src/IncrementalFetchingHelper.php

```diff
@@ -105,7 +148,56 @@ public function processItem(object $item, int $rowIndex): void
             ));
         }
 
-        $this->maxValue = max($this->maxValue, $newValue);
+        // Check if is type allowed, for the first row
+        if (!$this->valueTypeChecked) {
+            if (!in_array($newType, self::ALLOWED_TYPES, true)) {
+                throw new UserException(sprintf(
+                    'Unexpected type "%s" of the incremental fetching "%s" key. Allowed types "%s".',
+                    $newType,
+                    $this->key,
+                    implode('", "', self::ALLOWED_TYPES)
+                ));
+            }
+
+            if ($newType === EdmType::STRING) {
+                $this->logger->warning(sprintf(
+                    'Warning: Key "%s" - type "%s" is used for incremental fetching. ' .
+                    'For string type, all values must be the same length, ' .
+                    'otherwise incremental fetching fails.',
+                    $this->key,
+                    $newType
+                ));
+            }
+
+            $this->valueTypeChecked = true;
+        }
+
+        // Check if the string values have same length
+        // it's because of how string comparison works.
+        // Let's have string values "1" ... "1000"
+        // - if we used a condition >= "80", database returns "9", "80"
+        // - if we used a condition >= "99", database returns "99", but "100" is missing
+        if ($newType === EdmType::STRING) {
```

Comment:
A tu mu to spadne, ak sa to stane, tj. ak nesedia dlzky retazcov:
Tu je vysvetlene preco je to problem: https://keboola.slack.com/archives/CQ1ASK06M/p1606891806026900